src/checkpointTrie.js

```diff
@@ -111,60 +118,50 @@ module.exports = class CheckpointTrie extends BaseTrie {
   createScratchReadStream (scratch) {
     const trie = this.copy()
     scratch = scratch || this._scratch
+    scratch = new DB(scratch._db)
     // Only read from the scratch
-    trie._getDBs = [scratch]
+    trie.db = scratch
     trie._scratch = scratch
     return new ScratchReadStream(trie)
   }
 
+  putRaw (key, val, cb) {
+    if (this.isCheckpoint) return this._overridePutRaw(key, val, cb)
+    this.db.put(key, val, cb)
+  }
+
```

Comment:
To put this together and to your question, just summing up to make sure I get everything right: Before `putRaw` was ignoring checkpoint mechanism and would directly write to the permanent `DB`. This PR changes this behavior and `putRaw` writes to the scratch DB. If one wants to get "the old" functionality, one would directly use the method from `DB` to perform this.

Is this correct?

If so, I very much like this change, since this makes it very much more transparent and should avoid false usage of the API.

Reply:
I think when this is moved to `DB` and we decide to remove the methods from the trie class it becomes much clearer that the functionality is actually working on the `raw` un-checkpointed DB (and, to make this even clearer, we can also state additionally in the JSDoc comments).

So I think we should be good with this for now, since we are not taking away functionality. If people have some need for raw methods on the scratch DB, we can always think this properly through and add this later.